ruby-gem/test-server/instrumentation-backend/src/sh/calaba/instrumentationbackend/actions/ipc/SendTcp.java

```diff
@@ -0,0 +1,70 @@
+package sh.calaba.instrumentationbackend.actions.ipc;
+
+import sh.calaba.instrumentationbackend.Result;
+import sh.calaba.instrumentationbackend.actions.Action;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by john7doe on 22/01/15.
+ */
+
+public class SendTcp implements Action {
+    private static Map<Integer, Socket> connections = new HashMap<Integer, Socket>();
+
+    @java.lang.Override
+    public Result execute(String... args) {
+        if(args.length < 2 || args.length > 3) {
+            return Result.failedResult("Wrong number of arguments");
```

Comment:
I think we should give a better error. Like "Wrong number of arguments (x for 2,3), expected port, command, [keepConnectionAlive]